app/lib/features/settings/pages/change_password.dart

```diff
@@ -0,0 +1,218 @@
+          ],
+          validator: (val) {
+            if (val == null || val.trim().isEmpty) {
+              return L10n.of(context).emptyOldPassword;
+            }
+            return null;
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _buildNewPasswordInputField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(L10n.of(context).newPassword),
+        const SizedBox(height: 10),
+        TextFormField(
+          controller: newPassword,
+          obscureText: !_newPasswordVisible,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).hintMessagePassword,
+            suffixIcon: IconButton(
+              icon: Icon(
+                _newPasswordVisible ? Icons.visibility : Icons.visibility_off,
+              ),
+              onPressed: () {
+                setState(() {
+                  _newPasswordVisible = !_newPasswordVisible;
+                });
+              },
+            ),
+          ),
+          inputFormatters: [
+            FilteringTextInputFormatter.deny(RegExp(r'\s')),
+          ],
+          validator: (val) {
+            if (val == null || val.trim().isEmpty) {
+              return L10n.of(context).emptyNewPassword;
+            }
+            return null;
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _buildConfirmPasswordInputField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        Text(L10n.of(context).confirmPassword),
+        const SizedBox(height: 10),
+        TextFormField(
+          controller: confirmPassword,
+          obscureText: !_confirmPasswordVisible,
+          decoration: InputDecoration(
+            hintText: L10n.of(context).hintMessagePassword,
+            suffixIcon: IconButton(
+              icon: Icon(
+                _confirmPasswordVisible
+                    ? Icons.visibility
+                    : Icons.visibility_off,
+              ),
+              onPressed: () {
+                setState(() {
+                  _confirmPasswordVisible = !_confirmPasswordVisible;
+                });
+              },
+            ),
+          ),
+          inputFormatters: [
+            FilteringTextInputFormatter.deny(RegExp(r'\s')),
+          ],
+          validator: (val) {
+            if (val == null || val.trim().isEmpty) {
+              return L10n.of(context).emptyConfirmPassword;
+            } else if (val.trim() != newPassword.text.trim()) {
+              return L10n.of(context).validateSamePassword;
+            }
+            return null;
+          },
+        ),
+      ],
+    );
+  }
+
+  void _changePassword(BuildContext context) async {
+    if (!formKey.currentState!.validate()) return;
+    EasyLoading.show(status: L10n.of(context).changingYourPassword);
+    try {
+      final client = ref.read(alwaysClientProvider);
+      await client.changePassword(oldPassword.text, newPassword.text);
```

Comment:
err... minor error in consistency. you are always checking the confirmation to be `.trim()` but now you are submitting them without `.trim()`... should be the same for all (though probably not a problem in practice because of the `\s` denial regexp):

```suggestion
      await client.changePassword(oldPassword.text.trim(), newPassword.text.trim());
```